etl/steps/__init__.py

```diff
@@ -593,6 +608,71 @@ def _get_step_module(self) -> types.ModuleType:
         return step_module
 
 
+@dataclass
+class GrapherNewStep(DataStep):
+    path: str
+    dependencies: List[Step]
+
+    def __init__(self, path: str, dependencies: List[Step]) -> None:
+        self.path = path
+        self.dependencies = dependencies
+
+    def __str__(self) -> str:
+        return f"grapher-new://{self.path}"
+
+    def is_dirty(self) -> bool:
+        if any(d.is_dirty() for d in self.dependencies):
+            return True
+
+        # dataset exists, but it is possible that we haven't inserted everything into DB
+        dataset = self._output_dataset
+
+        return fetch_db_checksum(dataset) != self.checksum_input()
```

Comment:
> I wonder if we shouldn't include some kind of export schema version number in the db model somewhere

Not sure what you mean by _...in the db model_. Do you mean having something like having `VERSION = 2022-09-12` in code and adding it to checksum calculation so that if `VERSION` changes checksum changes too? (Perhaps we can move it to its own issue?)

Reply:
Yeah, I suppose that grapher-new steps could add a versioning constant to the source checksum they use to say if they're out of date. If you make a significant change, you bump it and then everything registers as needing to be re-exported.